ml/ADCharts.cc

```diff
@@ -3,55 +3,199 @@
 #include "ADCharts.h"
 #include "Config.h"
 
-void ml::updateDimensionsChart(RRDHOST *RH,
-                               collected_number NumTrainedDimensions,
-                               collected_number NumNormalDimensions,
-                               collected_number NumAnomalousDimensions) {
-    static thread_local RRDSET *RS = nullptr;
-    static thread_local RRDDIM *NumTotalDimensionsRD = nullptr;
-    static thread_local RRDDIM *NumTrainedDimensionsRD = nullptr;
-    static thread_local RRDDIM *NumNormalDimensionsRD = nullptr;
-    static thread_local RRDDIM *NumAnomalousDimensionsRD = nullptr;
-
-    if (!RS) {
-        std::stringstream IdSS, NameSS;
+// [ml] chart priorities
+#define ML_CHART_PRIO_DIMENSIONS        39181
+#define ML_CHART_PRIO_ANOMALY_RATE      39182
+#define ML_CHART_PRIO_DETECTOR_EVENTS   39183
 
-        IdSS << "dimensions_on_" << localhost->machine_guid;
-        NameSS << "dimensions_on_" << localhost->hostname;
+// [netdata] chart priorities
+#define NETDATA_ML_CHART_PRIO_MACHINE_LEARNING_STATUS 890001
+#define NETDATA_ML_CHART_PRIO_METRIC_TYPES            890002
+#define NETDATA_ML_CHART_PRIO_TRAINING_STATUS         890003
 
-        RS = rrdset_create(
-            RH,
-            "anomaly_detection", // type
-            IdSS.str().c_str(), // id
-            NameSS.str().c_str(), // name
-            "dimensions", // family
-            "anomaly_detection.dimensions", // ctx
-            "Anomaly detection dimensions", // title
-            "dimensions", // units
-            "netdata", // plugin
-            "ml", // module
-            39183, // priority
-            RH->rrd_update_every, // update_every
-            RRDSET_TYPE_LINE // chart_type
-        );
-        rrdset_flag_set(RS, RRDSET_FLAG_ANOMALY_DETECTION);
-
-        NumTotalDimensionsRD = rrddim_add(RS, "total", NULL,
-                1, 1, RRD_ALGORITHM_ABSOLUTE);
-        NumTrainedDimensionsRD = rrddim_add(RS, "trained", NULL,
-                1, 1, RRD_ALGORITHM_ABSOLUTE);
-        NumNormalDimensionsRD = rrddim_add(RS, "normal", NULL,
-                1, 1, RRD_ALGORITHM_ABSOLUTE);
-        NumAnomalousDimensionsRD = rrddim_add(RS, "anomalous", NULL,
-                1, 1, RRD_ALGORITHM_ABSOLUTE);
+#define NETDATA_ML_CHART_PRIO_PREDICTION_USAGE        890004
```

Comment:
Considering the values I am seeing here, I suspect that these values were generated based  in values already present inside [this file](https://github.com/netdata/netdata/blob/master/collectors/all.h), if this is true, I suggest one of the following options:

- Define these values inside the file that gave the idea.
- Comment something inside the original file.

I believe that in both situation we will avoid future conflicts.